lupa/_lupa.pyx

```diff
@@ -1182,13 +1188,21 @@ cdef object py_from_lua(LuaRuntime runtime, lua_State *L, int n):
     elif lua_type == lua.LUA_TNUMBER:
         if lua.LUA_VERSION_NUM >= 503:
             if lua.lua_isinteger(L, n):
-                return lua.lua_tointeger(L, n)
+                integer = lua.lua_tointeger(L, n)
+                if maxint and integer >= -maxint and integer < maxint:
+                        return <int>integer
+                else:
+                        return integer
             else:
                 return lua.lua_tonumber(L, n)
         else:
             number = lua.lua_tonumber(L, n)
-            if number == <lua.lua_Integer>number:
-                return <lua.lua_Integer>number
+            integer = <lua.lua_Integer>number
+            if number == integer:
+                if maxint and integer >= -maxint and integer < maxint:
+                        return <int>integer
+                else:
+                        return integer
```

Comment:
This seems worth a helper function. Also, the cast must be `<long>`, not `<int>`.

Reply:
That's not present in the current version of _lupa.pyx. You can safely ignore this and squash all commits into one.

Reply:
I'm not sure I understand. Do you want me to review a dozen separate features, arbitrarily split across 54 commits, without looking at separate commits and commit chains?